src/Extensions/Script/Serialization/JavaScriptSerializer.cs

```diff
@@ -1,42 +1,234 @@
+    private int _recursionLimit;
+    private int _maxJsonLength;
+
+    public JavaScriptSerializer() : this(null) { }
+
+    public JavaScriptSerializer(JavaScriptTypeResolver resolver)
+    {
+        _typeResolver = resolver;
+        RecursionLimit = DefaultRecursionLimit;
+        MaxJsonLength = DefaultMaxJsonLength;
+    }
+
+    public int MaxJsonLength
+    {
+        get
+        {
+            return _maxJsonLength;
+        }
+        set
+        {
+            if (value < 1)
+            {
+                throw new ArgumentOutOfRangeException(AtlasWeb.JSON_InvalidMaxJsonLength);
+            }
+            _maxJsonLength = value;
+        }
+    }
+
+    public int RecursionLimit
+    {
+        get
+        {
+            return _recursionLimit;
+        }
+        set
+        {
+            if (value < 1)
+            {
+                throw new ArgumentOutOfRangeException(AtlasWeb.JSON_InvalidRecursionLimit);
+            }
+            _recursionLimit = value;
+        }
+    }
+
+    internal JavaScriptTypeResolver TypeResolver
+    {
+        get
+        {
+            return _typeResolver;
+        }
+    }
+
+    private Dictionary<Type, JavaScriptConverter> _converters;
+    private Dictionary<Type, JavaScriptConverter> Converters
+    {
+        get
+        {
+            _converters ??= new Dictionary<Type, JavaScriptConverter>();
+            return _converters;
+        }
+    }
+
+    [SuppressMessage("Microsoft.Usage", "CA2301:EmbeddableTypesInContainersRule",
+        MessageId = "Converters", Justification = Constant.CA1859)]
+    public void RegisterConverters(IEnumerable<JavaScriptConverter> converters)
+    {
+        ArgumentNullException.ThrowIfNull(converters);
+
+        foreach (JavaScriptConverter converter in converters)
+        {
+            IEnumerable<Type> supportedTypes = converter.SupportedTypes;
+            if (supportedTypes != null)
+            {
+                foreach (Type supportedType in supportedTypes)
+                {
+                    Converters[supportedType] = converter;
+                }
+            }
+        }
+    }
+
+    [SuppressMessage("Microsoft.Usage", "CA2301:EmbeddableTypesInContainersRule",
+        MessageId = "_converters", Justification = Constant.CA1859)]
+    internal JavaScriptConverter GetConverter(Type t)
+    {
+        if (_converters != null)
+        {
+            while (t != null)
+            {
+                JavaScriptConverter converter;
+                if (_converters.TryGetValue(t, out converter))
+                {
+                    return converter;
+                }
+                t = t.BaseType;
+            }
+        }
+        return null;
+    }
+
+    internal bool ConverterExistsForType(Type t, out JavaScriptConverter converter)
+    {
+        converter = GetConverter(t);
+        return converter != null;
+    }
+
+    public object DeserializeObject(string input)
```

Comment:
this one doesn't appear used right now and deserializing a no-type object should be discouraged. Let's remove it for now.